src/components/Admin/Authorization/EditCommunityCredentials.tsx

```diff
@@ -45,6 +46,12 @@ export const EditCredentials: FC<EditCredentialsProps> = ({ parentMembers, crede
     onError: handleError,
   });
 
+  const { data: usersInfo, loading: loadingUsers } = useUsersQuery({
+    fetchPolicy: 'cache-and-network',
+    skip: parentMembers != null,
+  });
+  const allUsers = usersInfo?.users || [];
```

Comment:
This part together with the filtration of `availableMembers` can be extracted in a hook and be reused. The hook can accept the parent and current members and return availalble members and loading. Keep in mind that with `cache-and-network` fetch policy the useUsersQuery will refetch data on every rerender. It happens behind the scene and after the data is fetched the cache will be updated automatically and the new results will be visible which may lead to appearing of new users in the list. You can set the fetch policy like this:
```javascript
const { loading, error, data } = useQuery(GET_DOGS, {
  fetchPolicy: "network-only",   // Used for first execution

  nextFetchPolicy: "cache-first" // Used for subsequent executions
});

```
Which will guarantee that first time the component is loaded the data will be fetch from the network, so it will be fresh and next rerenders will use the cached data.

Reply:
Even the `userWithCredentials` query can be put in the hook.